src/formatting.rs

```diff
@@ -60,21 +70,47 @@ pub(crate) fn format_inner(contents: &[u8], _config: &Config) -> Vec<u8> {
         trace!("shape contents: {:?}", &content);
 
         match shape {
-            FlatShape::String | FlatShape::Int | FlatShape::Nothing => out.extend(bytes),
+            FlatShape::Int | FlatShape::Nothing => out.extend(bytes),
+            FlatShape::StringInterpolation => {
+                out.extend(bytes);
+            }
             FlatShape::List | FlatShape::Record => {
                 bytes = trim_ascii_whitespace(bytes);
-                let printable = String::from_utf8_lossy(bytes).to_string();
-                trace!("stripped the whitespace, result: {:?}", printable);
                 out.extend(bytes);
             }
+            FlatShape::Block | FlatShape::Closure => {
+                bytes = trim_ascii_whitespace(bytes);
+                out.extend(bytes);
+            }
+            FlatShape::String => {
+                out.extend(bytes);
+                // if it'a string after a `def`, add a space before the `[`
+                if after_a_def {
+                    out.extend(b" ");
+                }
+            }
             FlatShape::Pipe => {
                 out.extend(b"| ");
             }
-            FlatShape::External | FlatShape::ExternalArg => {
+            FlatShape::InternalCall(declid) => {
+                trace!("Called Internal call with {declid}");
+                out = resolve_call(bytes, declid, out);
+                // make the def_name bool true if its a `def` keyword
```

Comment:
not sure i understand this comment, is it out of date?

Reply:
No, it's recent, it's actually one of the 3 last lines I wrote yesterday.
It explains the reason of the following line `99`
```rust
   after_a_def = declid == 5;
```
because `declid == 5` look mysterious without explanation

Reply:
aaaah i see :+1: 

maybe
```rust
after_a_def = declid == DeclId::Def;
```
would be easier to read?
i think it's the `5` that got me confused

Reply:
Yep! Thanks for this thoughtful review! ❤️

Reply:
Do we get rid of the comment too? Or is the comment helping?
To me is clear, I fear for the new person that wants to read this code

Reply:
maybe using `DeclId::Def` is enough?
as you know, i would say it is but ehe :wink:

Reply:
deleted!